haddock-api/src/Haddock/Interface/Create.hs

```diff
@@ -964,7 +965,8 @@ moduleExport thisMod dflags ifaceMap instIfaceMap expMod =
           Just iface -> return [ ExportModule (instMod iface) ]
           Nothing -> do
             liftErrMsg $
-              tell ["Warning: " ++ pretty dflags thisMod ++ ": Could not find " ++
+              tell [dislocatedErrMsg $
+                    "Warning: " ++ pretty dflags thisMod ++ ": Could not find " ++
```

Comment:
The `"Warning: " ++ pretty dflags thisMod ++ ` is superfluous. Let's also pass the `moduleLocation` here.